src/main/java/org/pentaho/di/trans/dataservice/optimization/OptimizationImpactInfo.java

```diff
@@ -76,7 +78,7 @@ public String getDescription() {
       .append( "Step:  " )
       .append( getStepName() )
       .append( "\n" );
-    if ( getErrorMsg().length() > 0 ) {
+    if ( StringUtil.isEmpty( getErrorMsg() ) ) {
       builder
```

Comment:
@mkambol  yes, but errorMsg  is initialized to an empty string  private String errorMsg = "";
and I think that the first option would be appropriate

if ( getErrorMsg() == null || getErrorMsg().length() > 0 )